### ISSUE TITLE
v0.14.3: panic: value is marked, so must be unmarked first

### ISSUE THREAD

**Issue by dancrumb** [hashicorp/terraform#27403]
### Terraform Version

```
0.14.3
```

### Terraform Configuration Files
Unfortunately, it's unclear which part of the configuration is causing this. However, it's apparent that this is a pervasive issue with the latest version of Terraform, so I'm adding this to give you more info.

### Debug Output
Full trace is here: https://gist.github.com/dancrumb/26053878427f2f626d2ca4f89d951d06

Here's the Panic output:
```
Terraform v0.14.3
Configuring remote state backend...
Initializing Terraform configuration...


!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
...
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
panic: value is marked, so must be unmarked first

goroutine 614 [running]:
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.1/cty/marks.go:123
github.com/zclconf/go-cty/cty.Value.AsBigFloat(0x2c5c280, 0xc00005442f, 0x2551bc0, 0xc000cf8400, 0x2c5c201)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.1/cty/value_ops.go:1269 +0x4f
github.com/zclconf/go-cty/cty/gocty.fromCtyNumber(0x2c5c280, 0xc00005442f, 0x2551bc0, 0xc000cf8400, 0x237acc0, 0xc001bc5058, 0x182, 0xc001fbb470, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.1/cty/gocty/out.go:124 +0x4d
github.com/zclconf/go-cty/cty/gocty.fromCtyValue(0x2c5c280, 0xc00005442f, 0x2551bc0, 0xc000cf8400, 0x22f8120, 0xc001bc5058, 0x16, 0xc001fbb470, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.1/cty/gocty/out.go:85 +0x965
github.com/zclconf/go-cty/cty/gocty.FromCtyValue(0x2c5c280, 0xc00005442f, 0x2551bc0, 0xc000cf8400, 0x22f8120, 0xc001bc5058, 0x2c5c280, 0xc00005442f)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.7.1/cty/gocty/out.go:43 +0x114
github.com/hashicorp/terraform/terraform.(*EvalValidateResource).validateCount(0xc002385bc0, 0x2c97980, 0xc000cddba0, 0x2c5ae00, 0xc0002ba780, 0xc0009bed80, 0x7f5b407707a8, 0x8)
	/home/circleci/project/project/terraform/eval_validate.go:523 +0x74c
github.com/hashicorp/terraform/terraform.(*EvalValidateResource).Validate(0xc001fbbbc0, 0x2c97980, 0xc000cddba0, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval_validate.go:339 +0x16cc
github.com/hashicorp/terraform/terraform.(*NodeValidatableResource).Execute(0xc0003d66f0, 0x2c97980, 0xc000cddba0, 0xc00003a004, 0x24aa880, 0x275b780)
	/home/circleci/project/project/terraform/node_resource_validate.go:55 +0x2f5
github.com/hashicorp/terraform/terraform.(*ContextGraphWalker).Execute(0xc00149ad00, 0x2c97980, 0xc000cddba0, 0x7f5b4060bd90, 0xc0003d66f0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph_walk_context.go:127 +0xbc
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x275b780, 0xc0003d66f0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph.go:59 +0x962
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc0005e51a0, 0x275b780, 0xc0003d66f0, 0xc0009dbb80)
	/home/circleci/project/project/dag/walk.go:387 +0x375
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/home/circleci/project/project/dag/walk.go:309 +0x1246
```

### Crash Output
This is running on Terraform cloud, so the crash.log file is unavailable

### Expected Behavior
No crash

### Actual Behavior
Crash

### Steps to Reproduce
`terraform plan`

### Additional Context
I'm opening this as I have yet to see an issue where `AsBigFloat` is the function causing the panic.

### References

- #27107
- #27220 
- #27329

**Comment by dancrumb** [comment]
None of my variables are defined as `sensitive` in my TF configuration, but I do have values defined as "Sensitive" in my Terraform Cloud instance

**Comment by dancrumb** [comment]
For the record, I have been able to work around this by taking some of the variables that were marked sensitive (but did not need to be) and marked them as not sensitive.

I am now able to work around this crash, but this bug warrants investigation.

**Comment by dancrumb** [comment]
I think that the relevant resource might be this one:

```terraform
locals {
  secrets = {
    "SQL_HOST"     = var.workspace == "local" ? "localhost" : aws_db_instance.db[0].address
    "SQL_DATABASE" = var.workspace == "local" ? "postgres" : aws_db_instance.db[0].name
    "SQL_USER"     = var.workspace == "local" ? "ahana" : aws_db_instance.db[0].username
    # ...
    # a bunch of values, some from resources, some form external variables
    # ...
  }
}

resource aws_ssm_parameter "param" {
  count = length(local.secrets)

  name      = "/ahana/${var.workspace}/${sort(keys(local.secrets))[count.index]}"
  type      = "SecureString"
  value     = lookup(local.secrets, sort(keys(local.secrets))[count.index], "")
  overwrite = true
}
```

**Comment by pselle** [comment]
Thanks @dancrumb, I was able to reproduce your issue here https://github.com/danieldreier/terraform-issue-reproductions/blob/master/27403/main.tf and will work on a fix! Terraform Cloud leverages 0.14's sensitive variables feature to apply that to your sensitive data in TF Cloud, which is why you've been impacted despite not changing your config. Will get a fix out shortly!
